csi.proto

```diff
@@ -301,6 +323,179 @@ message Volume {
   // the same attributes. This field is OPTIONAL and when present MUST
   // be passed to volume validation and publishing calls.
   map<string,string> attributes = 3;
+
+  // Specifies where (regions, zones, racks, etc.) the provisioned
+  // volume is accessible from.
+  // A plugin that returns this field must also set the
+  // ACCESSIBILITY_CONSTRAINTS plugin capability.
+  // An SP may specify multiple topologies to indicates the volume is
+  // accessible from multiple locations.
+  // COs may use this information along with the topology information
+  // returned by NodeGetId to ensure that a given volume is accessible
+  // from a given node when scheduling workloads.
+  // This field is OPTIONAL. If it is not specified, the CO MAY assume
+  // the volume is equally accessible from all nodes in the cluster and
+  // may schedule workloads referencing the volume on any available
+  // node.
+  //
+  // Example 1:
+  //   accessible_topology = {"region": "R1", "zone": "Z2"}
+  // Indicates a volume accessible only from the "region" "R1" and the
+  // "zone" "Z2".
+  //
+  // Example 2:
+  //   accessible_topology =
+  //     {"region": "R1", "zone": "Z2"},
+  //     {"region": "R1", "zone": "Z3"} 
+  // Indicates a volume accessible from both "zone" "Z2" and "zone" "Z3"
+  // in the "region" "R1".
+  repeated Topology accessible_topology = 4;
+}
+
+message TopologyRequirement {
+  // Specifies the list of topologies the provisioned volume MUST be
+  // accessible from.
+  // This field is OPTIONAL. If TopologyRequirement is specified either
+  // permitted or preferred or both MUST be specified.
+  // 
+  // If permitted is specified, the provisioned volume must be
+  // accessible from at least one of the permitted topologies.
+  // 
+  // Given
+  //   x = number of topologies provisioned volume can be accessed from
+  //   n = number of permitted topologies
```

Comment:
So `n` is the number of entries of the `permitted` field and thus gets chosen by the CO. But where does `x` come from?

The `x==n` case below implies that this is somehow a requirement for the CreateVolume and thus something chosen by the CO ("only give  me a volume if it is accessible in at least n topologies"), but there is no field for it.

Reply:
X is an attribute of a given volume. For most volumes X=1 (meaning they are not synchronously replicated). X > 1 means the volume is accessible from more then 1 topology, for example, more then 1 zone (probably due to synchronous replication).

Reply:
So x really means "number of topologies provisioned volume can be accessed from", as in the definition. But then the examples don't make sense to me: "if a volume *should* be accessible from two zones" implies that there is another number y that might be smaller than x:
```
x = number of topologies provisioned volume can be accessed from
y = number of topologies that the provisioned volume must be accessible from
n = number of permitted topologies
```

Or is the assumption that y is always equal to x?

Reply:
Ah interesting point @pohly. X = Y so I'll reword it to `topologies that the provisioned volume **must** be accessible from`